src/Carbon/Carbon.php

```diff
@@ -1576,6 +1576,50 @@ public function isLeapYear()
     }
 
     /**
+     * Determines if the instance is in the current year
+     *
+     * @return bool
+     */
+    public function isCurrentYear()
+    {
+        return $this->isSameYear(static::now($this->tz));
+    }
+
+    /**
+     * Checks if the passed in date is in the same year as the instance year.
+     *
+     * @param \Carbon\Carbon $dt The instance to compare with.
+     *
+     * @return bool
+     */
+    public function isSameYear(Carbon $dt)
+    {
+        return $this->format('Y') === $dt->format('Y');
+    }
+
+    /**
+     * Determines if the instance is in the current month
+     *
+     * @return bool
+     */
+    public function isCurrentMonth()
+    {
+        return $this->isSameMonth(static::now($this->tz));
+    }
+
+    /**
+     * Checks if the passed in date is in the same month as the instance month.
+     *
+     * @param \Carbon\Carbon $dt The instance to compare with.
+     *
+     * @return bool
+     */
+    public function isSameMonth(Carbon $dt)
+    {
+        return $this->format('m') === $dt->format('m');
```

Comment:
In make sense, but I think passing a bool `$ofSameYear` would be useful: so that we can check if this the same month of the same year.